### ISSUE TITLE
declareAtom generic error with union

### ISSUE THREAD

**Issue by belozer** [artalar/reatom#192]
https://codesandbox.io/s/nostalgic-rain-1zbll

```tsx
import { declareAction } from "@reatom/core";

const action1 = declareAction<boolean>();
const action2 = declareAction<"ok" | "err">();
const action3 = declareAction<number>();
const action4 = declareAction<string>();

action1(false); // TS Error
action2("ok"); // TS Error

action3(35); // OK
action4("sgsdgs"); // OK
```

**Comment by artalar** [comment]
[Fix](https://www.typescriptlang.org/play/index.html#code/C4TwDgpgBAggxsAlgewHYBVzQLxQM7ABOiqA5gFDmiSwIqoA8ACgIYgA2yLAJlLi6hAAaKJhoQAHsAipueWkjRic+IiVIA+PlADe5KFGoQAXKKz6oYNpx6nWHLt3IBfSpLDJCwQ1gX0AwoQQLMCeDBb2NtxCFspQktKy8vCKGL64BMRkFlq4ABQA2pGOALrxUjJyUAUArrIQAGYkENwlFgYA-FB5eQCUfFop9Ax13I3N0WaQGr3tUKZ5Vg62UMU8-diDdGjM1o4iyjOUVL4w2kNogcGhhAwAjBrkY3DsLEFQAG5vUCymMOQsPJ3WYnGgAIXO21QVxCYTuUAAPlAAEyPZ6vd5fQhQABGpjB5BxQN6QA) example

**Comment by artalar** [comment]
Fixed in `@reatom/core@1.0.0-rc18`
